extensions/2.0/Khronos/KHR_texture_float_bc6h/README.md

```diff
@@ -0,0 +1,219 @@
+
+## glTF Schema Updates
+
+The `KHR_texture_float_bc6h` extension is added to the `textures` object and specifies a `source` property that points to the index of the `image` which defines a reference to the KTX v2 image in VK_FORMAT_BC6H_UFLOAT_BLOCK.
+
+The following glTF will load `image.ktx2` in clients that support this extension, and otherwise fall back to `image.png`.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_float_bc6h"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_float_bc6h": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.png"
+        },
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+When used in the glTF Binary (GLB) format the `image` that points to the KTX v2 resource uses the `mimeType` value of `image/ktx2`.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_float_bc6h"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_float_bc6h": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with Basis Universal supercompression without a fallback, define `KHR_texture_float_bc6h` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_float_bc6h"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_float_bc6h"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_float_bc6h": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_float_bc6h.schema.json](schema/texture.KHR_texture_float_bc6h.schema.json)
+
+## KTX v2 Images with VK_FORMAT_BC6H_UFLOAT_BLOCK
+
+To cover usecases where a texture source shall have increased dynamic range
+
+- Swizzling metadata (`KTXswizzle`) MUST be `rgba` or omitted.
+- Orientation metadata (`KTXorientation`) MUST be `rd` or omitted.
+- Color space information in the DFD MUST match the expected usage, namely:
+  - For textures with **color data** (e.g., base color maps),
+    - `colorPrimaries` MUST be `KHR_DF_PRIMARIES_BT709`;
```

Comment:
For material textures, there's virtually no benefit in using high-precision textures with the standard (aka bt.709) gamut. Bringing WCG material texture support to glTF would require much more infrastructure updates than adding a new pixel format.

Reply:
AFAIK the BT709 standard supports a bitdepth of 8 or 10 - could you please explain why this would not be beneficial in our usecase?

Reply:
Because to get that benefit in rendering context, all these specific conditions should be met:
* Original texture source data is authored with precision greater than _8-bit sRGB_, so there is actual data in those extra bits.
  * Note that _8-bit sRGB_ has roughly 12 bits of linear precision for values near zero and about 7 bits of linear precision for values near one.
* That extra precision is amplified enough during shading so that it makes a difference on a standard screen after the tone mapping.
  * A good example would be a simple linear gradient that is stretched over a very long object so that a single `1/255` step takes lots of screen space thus making that gradient look quantized.